### PR TITLE
Enable ImGui UI globally, remove legacy popup and keybindings UI

### PR DESCRIPTION
**Pull request by katemonster33**
#### Summary
Infrastructure "Deleted the ImGui enable/disable toggle, turns on ImGui globally in CDDA"

#### Purpose of change
Now that ImGui screens have reached a state where critical bugs have been squashed, we can proceed with the actual migration to ImGui UI. 

#### Describe the solution
Axe the "Use ImGui?" option from the options menu, and clear out all the code related to it. All legacy code behind that option is removed, and all code now unused because of the removal have also been removed.

#### Describe alternatives you've considered
We could proceed with the migration by keeping the option, with keeping all the legacy UI around and having the option to switch back and forth. But in my opinion, this is causing confusion where people are still maintaining the legacy UI when we should be focused on building the ImGui infrastructure.

#### Testing
Load cataclysm, open the popup, open the keybindings UI, verify that both display properly with no critical errors.

### CODE REVIEW

**Comment by oosyrag** [comment]
I'd like to request to keep the option.

**Comment by katemonster33** [comment]
as i said i'm not really in favor of it. taking away the option incentivizes more people to get involved in ImGui instead of just turning it off imo. at the end of the day I can only do so much myself, if the team is serious about this migration then we need more people doing the work.

#72936 clears the last few big blockers I saw in the tiled build, if there's other things then issues need to be made.

**Comment by andrei8l** [comment]
> I'd like to request to keep the option (but turn it on by default to get feedback).

That might encourage people to just disable imgui instead of submitting bug reports for their issues.

File: src/popup.cpp [hunk 1]
```diff
@@ -120,7 +123,7 @@ void query_popup_impl::on_resized()
                 for( const auto &opt : line ) {
                     button_width += get_text_width( remove_color_tags( opt ) );
                 }
-                // Right align.
+                // Righzt align.
```
Comment (dseguin) [inline src/popup.cpp:126]:
```suggestion
                // Right align.
```

Comment (katemonster33) [reply src/popup.cpp:126]: this is what i get for coding with a bird walking all over my keyboard

**Comment by AcidAntOnAMinefield** [comment]
> > I'd like to request to keep the option (but turn it on by default to get feedback).
> 
> That might encourage people to just disable imgui instead of submitting bug reports for their issues.

One issue is that it's ugly, how would that be fixed?

**Comment by gitelitko** [comment]
Is it possible to make it obey custom font settings? I play on a 1440p screen, and the default font is way too thin and so I use custom font, with custom size and spacing, but for some reason they don't apply to ImGui windows.

**Comment by katemonster33** [comment]
> One issue is that it's ugly, how would that be fixed?

Formulate your complaints into something actionable and write an issue? Or you could shout into the void in closed pull requests.

**Comment by katemonster33** [comment]
> Is it possible to make it obey custom font settings? I play on a 1440p screen, and the default font is way too thin and so I use custom font, with custom size and spacing, but for some reason they don't apply to ImGui windows.

Right now only the font height is obeyed by imgui, we're still learning how to work around it's inflexibility in this regard. Please write an issue, preferably with a side by side image showing font rendering using a legacy gui vs how they appear in imgui. Also a copy of your config.

**Comment by Drew4484** [comment]
> > > I'd like to request to keep the option (but turn it on by default to get feedback).
> > 
> > 
> > That might encourage people to just disable imgui instead of submitting bug reports for their issues.
> 
> One issue is that it's ugly, how would that be fixed?

This is unhelpful feedback.

**Comment by I-am-Erk** [comment]
Issues with imgui belong in issue posts; comments on merged/closed PRs are generally lost, so any useful feedback belongs elsewhere, and any pointless and unactionable feedback belongs in your own head.

**Pull request merged**